### DISCUSSION TITLE
Confusion in `struct View` in mesh_view_types

### DISCUSSION THREAD

**Discussion by unknownue** [bevyengine/bevy#5750] [Q&A]
The following code shows different ways to compute clip position in vertex shader:
```shader
#import bevy_pbr::mesh_view_types

@vertex
fn vertex(in: VertexInput) -> VertexOutput {
    ...
    out.clip_position = view.view_proj * world_position;  // Method 1
    out.clip_position = view.projection * view.view * world_position;  // Method 2
    out.clip_position = view.projection * view.inverse_view * world_position;  // Method 3
    return out;
}
```
It use the build-in shader struct defined in https://github.com/bevyengine/bevy/blob/681c9c6dc859be52565609e56ce9dc7f6f1c1f12/crates/bevy_pbr/src/render/mesh_view_types.wgsl#L3-L13

In practice, the Method 1 and Method 3 are work, and Method 2 is not work.
It seems that the function of `view.view` and `view.inverse_view` is not consistent with their naming.
Can someone explain this to me?

**Comment by superdump** [comment]
Transformations are applied to meshes. Think of your head as a camera, if you turn it to the left, that is like applying a view transform that rotates it to the left. Now bring your head back to the centre and hold your hand up in front of your face. Turn your head to the left again. Your hand was in front of your face, now it is to the right of your face. So from the point of view of the camera, rotating your head to the right (multiplying the camera position by the view transform) looks the same as rotating your hand to the right relative to your head. This is equivalent to applying the inverse view transform to the model’s vertex world position. The view_proj matrix is projection * inverse_view.

**Reply by unknownue** [reply]
It seems that the `view: mat4x4<f32>` member defines as "the transformation applied to camera" in Bevy, which is opposite to "the matrix that transforms vertices from world-space to view-space".
I am not expecting to argue which definition is accurate.
Including some comment/docs to it may help to avoid confusion.
